akka-cluster-bootstrap/src/main/scala/com/lightbend/rp/akkaclusterbootstrap/ClusterStatusCheck.scala

```diff
@@ -23,15 +23,22 @@ import scala.concurrent.{ ExecutionContext, Future }
 
 class ClusterStatusCheck extends ReadinessCheck with HealthCheck {
   def healthy(actorSystem: ExtendedActorSystem)(implicit ec: ExecutionContext): Future[Boolean] =
-    isUp(actorSystem)
+    status(actorSystem) map {
+      case MemberStatus.Joining  => true
+      case MemberStatus.WeaklyUp => true
+      case MemberStatus.Up       => true
+      case MemberStatus.Leaving  => false
+      case MemberStatus.Exiting  => false
+      case MemberStatus.Down     => false
+      case MemberStatus.Removed  => false
+    }
 
   def ready(actorSystem: ExtendedActorSystem)(implicit ec: ExecutionContext): Future[Boolean] =
-    isUp(actorSystem)
+    status(actorSystem) map (_ == MemberStatus.Up)
```

Comment:
If this becomes a k8s readiness check makes sense

Reply:
Nothing's changed here (it's just refactoring)

Reply:
Maybe WeaklyUp can also be considered ready though.
It's a detail since not that often used but seems logical I think.